### PR TITLE
Move virtio_net driver to common location for use on non-pc99 platforms.

### PR DESCRIPTION
**Pull request by kent-mcleod**
This driver can be used on platforms such as the qemu-arm-virt platform provided that the PCI bus has been initialized and the ioport BAR can be used via the ps_io_ops_t ioport interface.

### CODE REVIEW

File: libethdrivers/src/virtio_pci.c [hunk 1]
```diff
@@ -322,12 +322,12 @@ static int raw_tx(struct eth_driver *driver, unsigned int num, uintptr_t *phys,
...
     dev->tx_lengths[dev->tdt] = num;
     /* ensure update to descriptors visible before updating the index */
-    asm volatile("mfence" ::: "memory");
+    __atomic_thread_fence(__ATOMIC_SEQ_CST);
     dev->tdt = (dev->tdt + num + 1) % dev->tx_size;
     dev->tx_remain -= (num + 1);
     dev->tx_ring.avail->idx++;
     /* ensure index update visible before notifying */
-    asm volatile("mfence" ::: "memory");
+    __atomic_thread_fence(__ATOMIC_SEQ_CST);
```
Comment (kent-mcleod) [inline libethdrivers/src/virtio_pci.c:330]: This maps to asm volatile ("mfence" ::: "memory")

File: libethdrivers/src/virtio_pci.c [hunk 2]
```diff
@@ -262,9 +262,9 @@ static void fill_rx_bufs(struct eth_driver *driver) {
             .next = 0
         };
         dev->rx_ring.avail->ring[dev->rx_ring.avail->idx % dev->rx_size] = dev->rdt;
-        asm volatile("sfence" ::: "memory");
+        __atomic_thread_fence(__ATOMIC_RELEASE);
         dev->rx_ring.avail->idx++;
-        asm volatile("sfence" ::: "memory");
+        __atomic_thread_fence(__ATOMIC_RELEASE);
```
Comment (kent-mcleod) [inline libethdrivers/src/virtio_pci.c:267]: This maps to `asm volatile ("" ::: "memory")` which I believe should still be correct due to x86's memory semantics

Comment (oliver-wm) [reply libethdrivers/src/virtio_pci.c:267]: tested it?

**Comment by oliver-wm** [comment]
LGTM! Does the build system need to be updated somewhere?

**Review by oliver-wm — APPROVED** [review]

File: libethdrivers/src/virtio_pci.c (hunk 2 above)
Comment (kent-mcleod) [reply libethdrivers/src/virtio_pci.c:267]: yea

**Comment by kent-mcleod** [comment]
> LGTM! Does the build system need to be updated somewhere?

The CMake source glob expressions pick it up in the common src/ folder now.

**Comment by oliver-wm** [comment]
@ssrg-bamboo `test`

**Comment by ssrg-bamboo** [comment]
Hello, I'm a bot! I'll bring this PR into Trustworthy Systems and run some tests

**Comment by ssrg-bamboo** [comment]
All the tests we ran have passed! Nice job!

**Pull request merged**